### ISSUE TITLE
Firefox 57 and Webextensions

### ISSUE THREAD

**Issue by Endor8** [Baton34/organize-search-engines#109]
Hi.
I think you know from Firefox 57 on only web-extensions are allowed.
I am absolutely disappoint that the support of the so called legacy add-ons
will be removed by Mozilla. Have you any idea how and if this fantastic add-on
will survive. Do you see any chances to port it as web-extension?
I think the actually restrictions of web-extensions make it near to impossible to port it.
What can you say about it.
Cheers
Endor

**Comment by Baton34** [comment]
Hi.
I very disappointed too because i use some of addons that I'm 100% sure will not ported to web-extension.
I have no skills to port it to web-extension.
It fantastic addons will totally die in FF57 with no chances.
So you are need to make choice: modern FF (or other browser) or useful FF.
Migrate to esr branch. I suppose it will work with legacy addons until march 2018.

**Comment by Endor8** [comment]
Yea I think the only thing we can actually do is to migrate to ESR branch.
I have prepared the Firefox profile from Firefox 54.0 for this.
Since the profile from Firefox 55 is not backward compatible.
I don't understand why Mozilla takes such incomprehensible decision.
Mozilla apparently don't care about there users. 
Thanks for you fantastic work.
Cheers
Endor

**Comment by yfdyh000** [comment]
require https://bugzilla.mozilla.org/show_bug.cgi?id=1268401

**Comment by sorapyon** [comment]
If the API simply doesn't allow you to access the browser's search engines, why don't you use keyword search bookmarks/bookmarked searches/Smart keywords instead? Just as this extension does: [Context Search WebExtension](https://addons.mozilla.org/en-US/firefox/addon/context-search-we/)

You said that you don't have skills to port it to web-extension, but maybe you can rethink the extension taking into account keyword search bookmarks.

The drawback is that you can't use the "post" function (and favicons aren't supported), but as "search engines" are bookmarks, you can synchronize easily them between computers and they also can be organized in folders.

As translator, this extension is absolutely indispensable to me.

Thanks

**Comment by bege10** [comment]
I have switched to ESR already.
And today I found "Waterfox" www.waterfoxproject.org . Have a look at it. It will continue supporting non-WE addons and will have an own addon store.

**Comment by FreePaul** [comment]
Hi!
Look at Selection Context Search:
https://addons.mozilla.org/ru/firefox/addon/selection-context-search/?src=api

It is possible to make and organize your own search engines like in OSE. But only for popup menu.
Look at my own screenshot:
![default](https://user-images.githubusercontent.com/21068457/32833905-3b3f862c-ca4c-11e7-9eb0-d503daacbc84.png)

**Comment by Baton34** [comment]
Не факт, что в webext есть возможность реализовать какое-то подобие папок, а даже если и так, то доступа к панели поиска всё равно нет. А без этого это недоOSE получится. Да и потом его надо с нуля переписывать под WE, а у меня таких познаний нет.
В общем или OSE или модный FF.
Может быть кто-то другой когда-нибудь сможет написать что-то подобное OSE под we.

**Comment by MysteryIII** [comment]
Hello Baton34,
first of all I want to thank you for maintaining this great Add-On.
Are you familiar with UserChrome scripts? They are Userscripts that change the behavior of Firefox itself unlike GM Scripts that change the behavior of Web pages.
Here you find some additional inforamtions about it:

**Comment by LeeBinder** [comment]
I am with @MysteryIII on his suggestion.

**Comment by MysteryIII** [comment]
This is not an exact replace, but also brings back the old searchbar with some additional features:
https://github.com/Aris-t2/CustomJSforFx/blob/master/scripts/alternative_searchbar.uc.js

**Comment by LeeBinder** [comment]
thanks @MysteryIII Just use it as-is, or did you do manually change any settings ([un]comment any lines, toggle false/ true) of the script to have it be as close to OSE as possible?

Is there any way to get the groups back?

**Comment by bege10** [comment]
> This Add-on was written as a Firefox 57+ web-extensions compatible replacement for Ben Basson's popular Context Search Add-on

https://addons.mozilla.org/de/firefox/addon/contextsearch-web-ext/

**Comment by LeeBinder** [comment]
@bege10 : yes, quite powerful add-on even incl. search groups. I checked it out before and did so again just now. It's still missing the "Search with" contextual menu when highlighting a term in the search box, at least for me. Does that work for you, or doesn't either? (yes, I enabled "Enable on textarea and input elements")

**Comment by bege10** [comment]
> @bege10 : yes, quite powerful add-on even incl. search groups. I checked it out before and did so again just now. It's still missing the "Search with" contextual menu when highlighting a term in the search box, at least for me. Does that work for you, or doesn't either? (yes, I enabled "Enable on textarea and input elements")

I think I don't right understand the feature you mention. The add-on has its own search box. There I can enter a search item and then click the requested search engine. There I don't need a contextual menu. And in text areas and input elements the menu is present. Can you, please, explain more detailed what you miss?

**Comment by LeeBinder** [comment]
the feature you describe is a good start but for me does not quite cut it/ isn't equal to this OSE extension. It would be great to have a contextual menu sensitive text area right in the bar. Right now the search input field is not:

![Screen Shot 2020-05-29 at 23 13 56](https://user-images.githubusercontent.com/39203497/83306171-610b8280-a202-11ea-8012-33d632e9e7ea.jpg)

With the right coding/ scripting knowledge, I'm sure it would be possible to merge legacy OSE, Context search web ext. and the JS scripts into one combined web ext. with (close to) 100% the look, feel & functionality of OSE.

@Baton34 ?

**Comment by Baton34** [comment]
As I said before, I don’t have the necessary skills to write a new extension.

**Comment by LeeBinder** [comment]
@Baton34 : OK, gotcha - thanks for replying anyway.

At some point, someone will be able to pull on all strings and put all the pieces together. I'm happy with Waterfox Classic anyway.

**Comment by MysteryIII** [comment]
> 
> 
> thanks @MysteryIII Just use it as-is, or did you do manually change any settings ([un]comment any lines, toggle false/ true) of the script to have it be as close to OSE as possible?
> 
> Is there any way to get the groups back?

All customization options are in line 34-46. I've activated: 
old_search_engine_selection_popup
show_search_engine_names

Sadly there is no option for groups yet. But I think it might be possible. You might ask for it on a Firefox scripts board like [Camp Firefox](https://www.camp-firefox.de/forum/forum/16-individuelle-anpassungen/) or [Reddit](https://www.reddit.com/r/FirefoxCSS/).


I was staying on Fx 56 for very long time myself. Losing all the extensions was quite the problem and it took some time to get necessary APIs for quite a lot of features. While there's still a need here and there, we are at a point where it is pretty good again.
And now with the CSS Codes and scripts, I got almost everything I wished for.
You shouldn't forget that the new Firefox is far more performant and overall loading (itself, websites, scripts) is much faster.

**Comment by LeeBinder** [comment]
gotcha @MysteryIII , and thanks for the links.

I've been using Waterfox Classic. But the security patch gap is getting bigger, so I'm aiming at using Waterfox Current with CSS Codes and scripts including SE groups hopefully in the near future.

**Comment by Baton34** [comment]
Я сделал всё что мог и получилось то что есть. Не питайте больших иллюзий на поддержку и развитие так как я не программист и не имею должного уровня навыков и знаний. Можете попробовать новую версию 
https://github.com/Baton34/Organize-Search-Engines2/releases
I did my best and it turned out what it is. Do not have big illusions about support and development, since I am not a programmer and do not have the required level of skills and knowledge.
https://github.com/Baton34/Organize-Search-Engines2/releases

**Comment by LeeBinder** [comment]
very awesome, thank you Baton34, just tested your latest 0.0.7 and it's working fine (even when clicking onto SEs inside folder created in prefs) in latest FIrefox macOS! 🥇

**Comment by Baton34** [comment]
I asked "Classical Search Bar" author about folders and he said that have no time for this, so I am taked his addon as basis and create OSE2 and now he released a new version with folder support: https://addons.mozilla.org/ru/firefox/addon/classical-search-bar/ 
Try it.

**Comment by LeeBinder** [comment]
LOL but don't grudge - above all things are happening and advancing/ progressing - thanks for letting us know!

.. and honestly I do prefer YOUR way of implementing folders with the open-below style to 田生's because his is jumping around more and it's even necessary to click to go back, so please keep yours rather than copying his style. I'd say your style is more linear, elegant and user-friendly. Maybe you can look how he solved separators, but that's not something functional but only cosmetic and therefore low priority, I'd say.

**Comment by Baton34** [comment]
> but don't grudge

its not grudge its funy. I spent about 8 days for this cause I am a total noob in javascript, html and css.

**Comment by LeeBinder** [comment]
wow really - that I call commitment and devotion - kudos! It was obviously worth it, + you can use your new skills on other projects :)
